src/pages/signup.jsx

```diff
@@ -0,0 +1,93 @@
+                    }                         
+                    onFocus={()=>setErrorDetails("noerror")}
+                    required/> 
+
+                    <input type="text" id="lastname" className="input border-color-gray margin-none form-text" placeholder="Last Name"
+                     onChange={(e)=>setLastName(e.target.value)
+                    }                         
+                    onFocus={()=>setErrorDetails("noerror")}
+                    required/> 
+                    <input type="email" id="email" className="input border-color-gray margin-none form-text" placeholder="Email id"
+                    onChange={(e)=>setEmail(e.target.value)
+                    }                         
+                    onFocus={()=>setErrorDetails("noerror")}
+                    required/> 
+                    <input type="password" id="password" className="input border-color-gray margin-none form-text" placeholder="Password" 
+                    onChange={(e)=>setPassword(e.target.value)
+                    }                         
+                    onFocus={()=>setErrorDetails("noerror")}
+                    required/> 
+                    
+                 <div className="d-flex-cols align-items-center terms justify-content-between">
+                 <button className="btn btn-primary w-100 margin-none primary-color-bg border-radius-none"
+                               onClick={(e)=>{
+                                    e.preventDefault()
+                                    var validRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+                                    if(firstName!==""&&lastname!==""&&email!==""&&password!=="")
+                                    {
+                                       if(password.length>7)
+                                        {
+                                            if(email.match(validRegex))
+                                            {
+                                                     SignupHandler(firstName,lastname,email,password)
+                                                     setErrorDetails("noerror")
+                                            }
+                                            else{
+                                                setErrorDetails("Email should be valid")
+                                            }
+                                        }
+                                        else{
+                                            setErrorDetails("Password Should have atleast 8 characters")
+                                        }
+                                    }
+                                    else{
+                                        setErrorDetails("Fileds cannot be empty")
+                                    }
+                            
+                            }}
+                 >Submit</button>
```

Comment:
You can separate this logic in some handler function for easy readability

Reply:
Done Changes